packages/api/src/external/ehr/athenahealth/command/get-patients-from-appointments.ts

```diff
@@ -28,136 +28,133 @@ type PatientAppointment = {
 
   const patientAppointments: PatientAppointment[] = [];
   const getAppointmentsErrors: string[] = [];
-  const athenaClientKey = await getSecretValueOrFail(athenaClientKeySecretArn, region);
-  const athenaClientSecret = await getSecretValueOrFail(athenaClientSecretSecretArn, region);
+  const getAppointmentsArgs = cxMappings.map(mapping => {
+    const practiceId = mapping.externalId;
+    const departmentIds = mapping.secondaryMappings?.departmentIds;
+    if (departmentIds && !Array.isArray(departmentIds)) {
+      throw new Error(
+        `departmentIds exists but is malformed for cxId ${mapping.cxId} practiceId ${practiceId}`
+      );
+    }
+    return {
+      cxId: mapping.cxId,
+      practiceId,
+      departmentIds,
+      clientKey: athenaClientKey,
+      clientSecret: athenaClientSecret,
+      patientAppointments,
+      startLastModifiedDate: catchUp ? startLastModifiedDate : undefined,
+      endLastModifiedDate: catchUp ? endLastModifiedDate : undefined,
+      errors: getAppointmentsErrors,
+      log,
+    };
+  });
 
-  const now = new Date();
-  const currentDatetime = new Date(
-    Date.UTC(
-      now.getUTCFullYear(),
-      now.getUTCMonth(),
-      now.getUTCDate(),
-      now.getUTCHours(),
-      now.getUTCMinutes(),
-      now.getUTCSeconds()
-    )
-  );
-  const startAppointmentDate = new Date(currentDatetime);
-  const endAppointmentDate = new Date(
-    new Date(currentDatetime).setUTCFullYear(
-      currentDatetime.getUTCFullYear() + appointmentYearsLookForward
-    )
-  );
-  const startLastModifiedDate = new Date(
-    new Date(currentDatetime).setUTCHours(currentDatetime.getUTCHours() - lastModifiedHoursLookback)
-  );
-  const endLastModifiedDate = new Date(currentDatetime);
-  log(`Getting appointments from ${startLastModifiedDate} to ${endLastModifiedDate}`);
-
-  await executeAsynchronously(
-    cxMappings.flatMap(mapping => {
-      const cxId = mapping.cxId;
-      const practiceId = mapping.externalId;
-      const departmentIds = mapping.secondaryMappings?.departmentIds;
-      if (!departmentIds || !Array.isArray(departmentIds) || departmentIds.length === 0) {
-        log(`Skipping for cxId ${cxId} -- departmentIds missing, malformed or empty`);
-        return [];
-      }
-      return departmentIds.map(departmentId => {
-        return {
-          cxId,
-          practiceId,
-          departmentId,
-          clientKey: athenaClientKey,
-          clientSecret: athenaClientSecret,
-          patientAppointments,
-          startAppointmentDate,
-          endAppointmentDate,
-          startLastModifiedDate,
-          endLastModifiedDate,
-          errors: getAppointmentsErrors,
-          log,
-        };
-      });
-    }),
-    getAppointmentsAndCreateOrUpdatePatient,
-    { numberOfParallelExecutions: 10, delay: delay.asMilliseconds() }
-  );
+  await executeAsynchronously(getAppointmentsArgs, getAppointmentsFromSubAndCreateOrUpdatePatient, {
+    numberOfParallelExecutions: 2,
```

Comment:
nit: magic number. Why 2?

Reply:
No particular reason -- just rate limiting how many practices we run in parallel. Made a variable.